pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -65,6 +68,14 @@ func (configgen *ConfigGeneratorImpl) BuildClusters(env *model.Environment, prox
 		if configgen.CanUsePrecomputedCDS(proxy) {
 			if sidecarScope != nil && sidecarScope.XDSOutboundClusters != nil {
 				clusters = append(clusters, sidecarScope.XDSOutboundClusters...)
+				// For locality loadbalancing
+				for i, cluster := range clusters {
+					if cluster.LoadAssignment != nil {
+						clone := util.CloneCluster(cluster)
+						ApplyLocalityLBSetting(proxy, cluster, push)
+						clusters[i] = clone
+					}
```

Comment:
this is fine for the first cut. But this will make the cache pointless. In a subsequent PR I suggest the following optimization to get cache benefits:
after you call buildLocalityLbEndpoints, obtain the list of all localities (region/zone/subzone). Each cluster in the pre-generated CDS output should have N copies in the SidecarScope object - one copy per locality or the default copy if its an eds cluster.
Costin is planning to use the same object to store EDS cache per locality

Reply:
sounds good to me.

Reply:
Oh I just read your earlier response. If you think caching here is not useful, add a comment stating that reason.. helpful in future.

Reply:
done